app/Models/BuildFailure.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Database\Eloquent\Relations\HasOne;
+
+/**
+ *
+ * @property int $buildid
+ * @property int $detailsid
+ * @property string $sourcefile
+ * @property int $newstatus
+ *
+ * @mixin Builder<BuildFailure>
+ */
+class BuildFailure extends Model
```

Comment:
I think we decided to call these "rich build alerts" in the past, to differentiate between "basic" and "rich" types without implying that warnings are stored elsewhere.